src/main/java/software/amazon/documentdb/jdbc/DocumentDbAbstractResultSet.java

```diff
@@ -282,8 +282,11 @@ public ResultSetMetaData getMetaData() {
 
     @Override
     public Object getObject(final int columnIndex) throws SQLException {
-        // Don't try to convert to Object here.
-        return getValue(columnIndex);
+        // Don't try to convert to Object here. Return value as is but check if null.
+        verifyState(columnIndex);
+        final Object o = getValue(columnIndex);
+        wasNull = (o == null);

```

Comment:
wasNull flag was not being set for getObject since it doesn't go through getValue(columnIndex, type) method.

Reply:
Good catch.